File: URI/Python/1047.py
hi, mi, hf, mf = input().split()
hi = int(hi)
mi = int(mi)
hf = int(hf)
mf = int(mf)

horas = 0
minutos = 0

if hi == hf:
	if mi == mf:
		horas = 24
		minutos = 0
	elif mi < mf:
		horas = 0
		minutos = mf - mi
	elif mi > mf:
		horas = 23
		minutos = 60 - mi + mf
elif hi < hf:
	if mi == mf:
		horas = hf - hi
		minutos = 0
	else:
		horas = hf - (hi + 1)
		minutos = 60 - mi + mf
elif hi > hf:
	horas = 24 - (hi + 1) + hf
	minutos = 60 - mi + mf

if minutos >= 60:
	minutos -= 60
	horas += 1

print("O JOGO DUROU {} HORA(S) E {} MINUTO(S)".format(horas, minutos))

File: URI/C/1047.c
#include<stdio.h>
#include<stdlib.h>

int h1, m1, h2, m2, qtd_min = 0;

int main(){
	scanf("%d %d %d %d", &h1, &m1, &h2, &m2);
	if(h1 == h2){
		if(m1 == m2)
			printf("O JOGO DUROU 24 HORA(S) E 0 MINUTO(S)\n");
		else if(m1 < m2){
			qtd_min += m2 - m1;
			printf("O JOGO DUROU %d HORA(S) E %d MINUTO(S)\n", qtd_min/60, qtd_min%60);	
		}else if(m1 > m2){
			qtd_min += (23*60);
			qtd_min += m2 + (60 - m1); 
			printf("O JOGO DUROU %d HORA(S) E %d MINUTO(S)\n", qtd_min/60, qtd_min%60);
		}
	}else if(h1 < h2){
		qtd_min += 60 - m1;
		qtd_min += (h2 - (h1+1))*60;
		qtd_min += m2;
		printf("O JOGO DUROU %d HORA(S) E %d MINUTO(S)\n", qtd_min/60, qtd_min%60);
	}else if(h1 > h2){
		qtd_min += 60 - m1;
		qtd_min += m2;
		qtd_min += (24 - (h1 + 1))*60;
		qtd_min += h2*60;
		printf("O JOGO DUROU %d HORA(S) E %d MINUTO(S)\n", qtd_min/60, qtd_min%60);
	} 
	
	return 0;
}
